openapi/openapi.yml

```diff
@@ -860,6 +860,21 @@ paths:
           $ref: '#/components/responses/Error'
       tags:
         - Postgres Database
+  '/project/{project_id}/location/{location}/postgres/{postgres_database_name}/failover':
```

Comment:
I guess we lacked a test for this as well? I wonder how that works given 100%?

Reply:
There is an existing `openapi.yml` entry for `'/project/{project_id}/location/{location}/postgres/_{postgres_database_id}/failover'`, so I'm guessing coverage was handled by that.  Basically, there are four cases:

1. web - name
2. web - ubid
3. api - name
4. api - ubid

You can get 100% branch coverage when testing only three of these four cases.  However, the cli test I wrote uses `api - name`, which was previous untested and didn't work.

This type of issue may well affect other cases.  We could avoid this issue by not having separate entries in `openapi.yml` for the name and ubid cases.

Reply:
Do you know why we have two? I presume two distinct patterns for tighter validation or something? Seems like the costs may not justify so much bookkeeping, if this is the one thing causing spurious rejections of requests and doubling up schema entries, cc @geemus

Reply:
I do not know.  I'm also not sure why having the duplicate route works in this case, but broke tests when I converted all `/id/` routes to `/_` routes (I tried that approach, but since it didn't work, I submitted #2783 to remove the `/id/` routes).

Reply:
FWIW, I recommend that we accept either name or ubid (without underscore), and have a name validation that does not allow ubid format for the model's ubid type (chance of someone using ubid format by accident is minimal).  We would need to check that nobody is currently doing that, though.